src/ResourceManager/KeyVault/Commands.KeyVault/Commands/BackupAzureKeyVaultKey.cs

```diff
@@ -37,6 +39,7 @@ public class BackupAzureKeyVaultKey : KeyVaultCmdletBase
                    ValueFromPipelineByPropertyName = true,
                    HelpMessage = "Vault name. Cmdlet constructs the FQDN of a vault based on the name and currently selected environment.")]
         [ValidateNotNullOrEmpty]
+        [ValidatePattern(Constants.VaultNameRegExString)]
```

Comment:
This is potentially a breakign change - does this exclude any values that customers might have in their scripts which would have succeeded before?  Or would any values failign this check fail on the service anyway?